ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -1222,28 +1225,26 @@ private class JdbcLedgerDao(
       .toSet
 
   private val SQL_GET_LEDGER_ENTRIES = SQL(
-    "select * from ledger_entries where ledger_offset>={startInclusive} and ledger_offset<{endExclusive} order by ledger_offset asc")
+    "select * from ledger_entries where ledger_offset>={startInclusive} and ledger_offset<{endExclusive} order by ledger_offset asc limit {pageSize} offset {queryOffset}")
 
   // Note that here we are reading, non transactionally, the stream in chunks. The reason why this is
   // safe is that
   // * The ledger entries are never removed;
   // * We fix the ledger end at the beginning.
   private def paginatingStream[T](
```

Comment:
I'd love for this to be pulled out into its own file and wrapped in unit tests, mostly for documentation purposes. It's hard to grok just by reading the code.